### DISCUSSION TITLE
How can I add new property to object?

### DISCUSSION THREAD

**Discussion by pavankjadda** [salesforce/akita#723] [Q&A]
In my entity store I defined `employee` state and store with few attributes. 
**Employee.ts**
```
public interface Employee
{
 firstName: string;
 lastName: string;
 email: string;
 phone: string;
 ...
 employeeProjectCount: number;
}

```

Based on data received from backend I need to compute and update attribute like `employeeProjectsCount` and update the state. And this attribute is not defined in backend hence does not come as JSON response (However I defined it on TS Interface that maps to JSON data. See above) 

Here is the problem. When I compute and try to update the object I get error 

`cannot add property, object is not extensible` 

I use the following piece of code to update the object 

```
this.employeeService.getemployeeProjectCount(URL).subscribe(data=>
{
  employee={...employee, employeeProjectCount: data};
  this.employeeStore.update(employee.id, employee);
}

```

**Comment by NetanelBasal** [comment]
Where is the update call?

**Reply by pavankjadda** [reply]
See my updated code.

**Reply by NetanelBasal** [reply]
The error says that you are mutating the state. I guess you are mutating the `data` property in your case, which probably comes from the store.

**Comment by pavankjadda** [comment]
It's gone now. The `employee` object fetched from store and I am trying to update it. Copying it to temporary object and updating it fixed the error.
